### PR TITLE
add more dimension mapping types

### PR DESCRIPTION
**Pull request by lixiliu**
add mapping types and archetypes that check fraction_sum = 1 when grouped by to_id. Currently, `DimensionMappingArchetype` only supports  fraction_sum check when grouped by from_id.

Completes JIRA Story DSGRID- (part of effort to get AEO growth rates registered)
Closes GitHub Issue #

## Description
**"aggregation" & "disaggregation"** mapping types will have fraction_sum check grouped by **from_id**
**"assignment"** mapping types will have fraction_sum check grouped by **to_id**
AEO growth rate mapping will be "many_to_many_assignment": https://github.com/dsgrid/dsgrid-project-StandardScenarios/tree/ll/historical/dsgrid_project/datasets/sector_models/aeo2021_reference/commercial/End_Use_Growth_Factors/dimension_mappings

## Checklist

- [ ] Tests exercising the new feature or bug fix
- [ ] All tests pass
- [ ] At least one code review approval
- [ ] Consider transferring TODOs to JIRA or GitHub

### CODE REVIEW

File: dsgrid/config/dimension_mapping_base.py [hunk 1]
```diff
@@ -36,65 +38,108 @@ class DimensionMappingType(DSGEnum):
...
         value="many_to_many_map_fraction_sum_eq1",
         description="Many-to-many dimension mapping with sum of from_fraction = 1 when grouped by from_id",
         allow_dup_from_records=True,
         allow_dup_to_records=True,
-        check_fraction_sum_eq1=True,
+        check_fraction_sum_eq1_fromid=True,
+        check_fraction_sum_eq1_toid=False,
+    )
+
+    ONE_TO_ONE_MAP_FRACTION_SUM_EQ1_TOID = EnumValue(
```
Comment (daniel-thom) [inline dsgrid/config/dimension_mapping_base.py:79]: Can you make all of these `to_id`, `TOID`, `from_id`, `FROM_ID`?

File: dsgrid/config/dimension_mapping_base.py [hunk 2]
```diff
@@ -36,65 +38,108 @@ class DimensionMappingType(DSGEnum):
...
+        value="one_to_one_map_fraction_sum_eq1",
+        description="One-to-one dimension mapping with sum of from_fraction = 1 when grouped by to_id",
+        allow_dup_from_records=False,
+        allow_dup_to_records=False,
+        check_fraction_sum_eq1_fromid=False,
+        check_fraction_sum_eq1_toid=True,
+    )
+    ONE_TO_MANY_MAP_FRACTION_SUM_EQ1_TOID = EnumValue(
```
Comment (daniel-thom) [inline dsgrid/config/dimension_mapping_base.py:87]: Should we have symmetry with the names? Like
```
ONE_TO_MANY_MAP_FRACTION_SUM_EQ1_FROM_ID
ONE_TO_MANY_MAP_FRACTION_SUM_EQ1_TO_ID
```

File: dsgrid/config/dimension_mapping_base.py [hunk 3]
```diff
@@ -14,19 +14,21 @@
...
+
+    # required from_fraction col, FRACTION_SUM_EQ1 when grouped by to_id
     ONE_TO_MANY_DISAGGREGATION = "one_to_many_disaggregation"
     MANY_TO_MANY_DISAGGREGATION = "many_to_many_disaggregation"
 
-    # input from_fraction col
+    # required from_fraction col, no FRACTION_SUM check
     ONE_TO_ONE_EXPLICIT_MULTIPLIERS = "one_to_one_explicit_multipliers"
     ONE_TO_MANY_EXPLICIT_MULTIPLIERS = "one_to_many_explicit_multipliers"
     MANY_TO_ONE_EXPLICIT_MULTIPLIERS = "many_to_one_explicit_multipliers"
```
Comment (elainethale) [inline dsgrid/config/dimension_mapping_base.py:34]: Should there be MANY_TO_MANY_EXPLICIT_MULTIPLIERS?

File: dsgrid/config/dimension_mapping_base.py [hunk 4]
```diff
@@ -14,19 +14,21 @@
...
-    # default from_fraction col
+    # optional from_fraction col, no FRACTION_SUM check
     DUPLICATION = "duplication"
 
-    # input from_fraction col, FRACTION_SUM_EQ1
+    # required from_fraction col, FRACTION_SUM_EQ1 when grouped by from_id
     MANY_TO_MANY_AGGREGATION = "many_to_many_aggregation"
+
+    # required from_fraction col, FRACTION_SUM_EQ1 when grouped by to_id
     ONE_TO_MANY_DISAGGREGATION = "one_to_many_disaggregation"
```
[the remark this answers is not in the document]
Comment (elainethale) [reply dsgrid/config/dimension_mapping_base.py:28]: In conversation, agree that we should group by from_id

**Review by elainethale — COMMENTED** [review]
AEO growth rate mapping for end uses is something different--allocation, assignment, attribution. In that case, the sum after grouping by to_id should be 1.0.

File: dsgrid/registry/dimension_mapping_registry_manager.py [hunk 5]
```diff
@@ -162,9 +163,16 @@ def validate_records(self, config: DimensionMappingsConfig):
                 mapping.mapping_type.value,
             )
 
-            if mapping.archetype.check_fraction_sum_eq1:
+            if mapping.archetype.check_fraction_sum_eq1_from_id:
```
Comment (daniel-thom) [inline dsgrid/registry/dimension_mapping_registry_manager.py:166]: Does this change affect any mappings in the project repositories?

Comment (lixiliu) [reply dsgrid/registry/dimension_mapping_registry_manager.py:166]: no, this is only a name change. We've only checked sum group by from_id for all mappings so far.

**Review by elainethale — APPROVED** [review]

**Pull request merged**